conda/environments/all_cuda-118_arch-x86_64.yaml

```diff
@@ -11,12 +11,12 @@ dependencies:
 - cmake>=3.23.1,!=3.25.0
 - cuda-python>=11.7.1,<12.0
 - cudatoolkit=11.8
-- cudf=23.04.*
-- cupy>=7.8.0,<12.0.0a0
+- cudf==23.4.*
```

Comment:
Seems to be inconsistent versions here. Some RAPIDS libraries are `23.4` and some are `23.04` (eg line 47 for `librmm`). I think we should be consistent using the zero-padded versions for conda (though technically, `23.4.*` _does_ work).

Reply:
The issue here is that while conda supports zero-padded versions, newer versions of pip do not. More specifically, newer versions of packages in the Python packaging world take a stricter view of [PEP 440](https://peps.python.org/pep-0440/) than conda does. In particular, the [integer normalization rules](https://peps.python.org/pep-0440/#integer-normalization) do not allow for leading zeros.

There are other similar issues. One of the most common that I have run into is that with conda a single equals sign is a fuzzy match and a double equals sign is an exact match (e.g. `cudf=23.4` is equivalent to `cudf==23.4.*`, but `cudf==23.4` will not match `23.4.1`), but only `cudf==23.4.*` is accepted by newer versions of [packaging](https://github.com/pypa/packaging), the de facto standard for version parsing.

I don't know if conda has a plan to phase out support for older version specs (I would guess not anytime soon), but that doesn't really matter. In order to use dependencies.yaml as a single source of truth, are over time going to have to migrate to using the limited common subset of supported version syntaxes.

Reply:
Done.